src/Symfony/Component/Console/Attribute/Option.php

```diff
@@ -0,0 +1,119 @@
+use Symfony\Component\Console\Completion\Suggestion;
+use Symfony\Component\Console\Exception\LogicException;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+
+#[\Attribute(\Attribute::TARGET_PARAMETER)]
+class Option
+{
+    private const ALLOWED_TYPES = ['string', 'bool', 'int', 'float', 'array'];
+
+    private ?int $mode = null;
+    private string $typeName = '';
+
+    /**
+     * Represents a console command --option definition.
+     *
+     * If unset, the `name` and `default` values will be inferred from the parameter definition.
+     *
+     * @param array|string|null                                              $shortcut        The shortcuts, can be null, a string of shortcuts delimited by | or an array of shortcuts
+     * @param scalar|array|null                                              $default         The default value (must be null for self::VALUE_NONE)
+     * @param array|callable-string(CompletionInput):list<string|Suggestion> $suggestedValues The values used for input completion
+     */
+    public function __construct(
+        public string $name = '',
+        public array|string|null $shortcut = null,
+        public string $description = '',
+        public string|bool|int|float|array|null $default = null,
+        public array|string $suggestedValues = [],
+    ) {
+        if (\is_string($suggestedValues) && !\is_callable($suggestedValues)) {
+            throw new \TypeError(\sprintf('Argument 5 passed to "%s()" must be either an array or a callable-string.', __METHOD__));
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public static function tryFrom(\ReflectionParameter $parameter): ?self
+    {
+        /** @var self $self */
+        if (null === $self = ($parameter->getAttributes(self::class, \ReflectionAttribute::IS_INSTANCEOF)[0] ?? null)?->newInstance()) {
+            return null;
+        }
+
+        $type = $parameter->getType();
+        $name = $parameter->getName();
+
+        if (!$type instanceof \ReflectionNamedType) {
+            throw new LogicException(\sprintf('The parameter "$%s" must have a named type. Untyped, Union or Intersection types are not supported for command options.', $name));
+        }
+
+        $self->typeName = $type->getName();
+
+        if (!\in_array($self->typeName, self::ALLOWED_TYPES, true)) {
+            throw new LogicException(\sprintf('The type "%s" of parameter "$%s" is not supported as a command option. Only "%s" types are allowed.', $self->typeName, $name, implode('", "', self::ALLOWED_TYPES)));
+        }
+
+        if (!$self->name) {
+            $self->name = $name;
+        }
+
+        if ('bool' === $self->typeName) {
+            $self->mode = InputOption::VALUE_NONE | InputOption::VALUE_NEGATABLE;
+        } else {
+            $self->mode = null !== $self->default || $parameter->isDefaultValueAvailable() ? InputOption::VALUE_OPTIONAL : InputOption::VALUE_REQUIRED;
```

Comment:
this check looks wrong to me. `InputOption::VALUE_OPTIONAL` is hard to use right, as using it properly requires to distinguish 3 cases (which cannot be distinguished by just using the result of `$input->getOptions()`:

- option not passed at all (uses the default value when getting it)
- option passed without value (uses the default value when getting it)
- option passed with a value

We should probably always use `InputOption::VALUE_REQUIRED` here

Reply:
Fixed in https://github.com/symfony/symfony/pull/59493, thanks!